Mathlib/AlgebraicTopology/SimplicialObject/Coskeletal.lean

```diff
@@ -0,0 +1,116 @@
+
+TODO: Prove that `X` is `n`-coskeletal whenever a certain canonical cone is a limit cone.
+-/
+
+open Opposite
+
+open CategoryTheory
+
+open CategoryTheory.Limits CategoryTheory.Functor SimplexCategory
+
+universe v u v' u'
+
+namespace CategoryTheory
+
+namespace SimplicialObject
+variable {C : Type u} [Category.{v} C]
+variable (X : SimplicialObject C) (n : ℕ)
+
+namespace Truncated
+
+/-- The identity natural transformation exhibits a simplicial set as a right extension of its
+restriction along `(Truncated.inclusion n).op`.-/
+@[simps!]
+def rightExtensionInclusion :
+    RightExtension (Truncated.inclusion n).op
+      ((Truncated.inclusion n).op ⋙ X) := RightExtension.mk _ (𝟙 _)
+
+end Truncated
+
+open Truncated
+
+/-- A simplicial object `X` is `n`-coskeletal when it is the right Kan extension of its restriction
+along `(Truncated.inclusion n).op` via the identity natural transformation. -/
+class IsCoskeletal : Prop where
+  nonempty_isRightKanExtension :
+    Nonempty (IsRightKanExtension X (𝟙 ((Truncated.inclusion n).op ⋙ X)))
+
+variable [X.IsCoskeletal n]
+
+/-- If `X` is `n`-cosketal, then `𝟙 ((Truncated.inclusion n).op ⋙ X)` defines a right Kan
+extension of `(Truncated.inclusion.op ⋙ X)` along `(Truncated.inclusion n).op`. -/
+theorem IsCoskeletal.isRightKanExtension :
+    IsRightKanExtension X (𝟙 ((Truncated.inclusion n).op ⋙ X)) :=
+  IsCoskeletal.nonempty_isRightKanExtension.some
+
+/-- If `X` is `n`-coskeletal, then `rightExtensionInclusion X n` is a terminal object in the
+category `RightExtension (Truncated.inclusion n).op (Truncated.inclusion.op ⋙ X)`. -/
+noncomputable def IsCoskeletal.isUniversalOfIsRightKanExtension :
+    (rightExtensionInclusion X n).IsUniversal := by
+  have := isRightKanExtension X n
+  apply Functor.isUniversalOfIsRightKanExtension
+
+variable [∀ (F : (SimplexCategory.Truncated n)ᵒᵖ ⥤ C),
+    (SimplexCategory.Truncated.inclusion n).op.HasRightKanExtension F]
+
+/-- There is a map of costructured arrows from `rightExtensionInclusion X n` to the right extension
+of the `n`-truncation of `X` defined by the counit of `coskAdj n`. -/
+noncomputable def Truncated.rightExtensionCosk.hom : Truncated.rightExtensionInclusion X n ⟶
+    RightExtension.mk _
+      ((coskAdj n).counit.app ((Truncated.inclusion n).op ⋙ X)) :=
+  CostructuredArrow.homMk ((coskAdj n).unit.app X) ((coskAdj n).left_triangle_components X)
+
+instance Truncated.isRightKanExtensionCosk : IsRightKanExtension
+    ((Truncated.cosk n).obj ((inclusion n).op ⋙ X))
+    ((coskAdj n).counit.app ((inclusion n).op ⋙ X)) := by
+  unfold Truncated.cosk coskAdj
+  rw [ranAdjunction_counit]
+  exact Functor.instIsRightKanExtensionObjRanAppRanCounit _ _
+
+/-- The map `coskRightExtension.hom X` is a natural transformation between two right Kan extensions
+of the diagram `Truncated.inclusion.op ⋙ X` and thus is an isomorphism. -/
+instance IsCoskeletal.coskRightExtension.isIsoHom : IsIso (rightExtensionCosk.hom X n) :=
```

Comment:
```suggestion
instance IsCoskeletal.coskRightExtension.isIso_hom : IsIso (rightExtensionCosk.hom X n) :=
```
(`IsIso` is a Prop, so that this instance is like a lemma, not a def, then we have to use `_` as a separator.)

Reply:
The name should rather contain `rightExtensionCosk` rather than `coskRightExtension`. Then, better names would be `IsCoskeletal.isIso_rightExtensionCosk_hom` (or `isIso_rightExtensionCosk_hom_of_isCoskeletal`).